CMakeLists.txt

```diff
@@ -1,16 +1,24 @@
 cmake_minimum_required(VERSION 3.5)
 project(LibreCuda CXX)
 
-# create elf-parser library
-add_subdirectory(third_party/ELFIO)
+set( CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin )
+set( CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib )
+
+include_directories(
+        ${CMAKE_SOURCE_DIR}/src
+)
 
-add_executable(LibreCuda
-        src/main.cpp
-        src/librecuda.cpp
-        src/librecuda_status.cpp
-        src/cmdqueue.cpp)
+if(NOT EXISTS ${CMAKE_SOURCE_DIR}/third_party/ELFIO/elfio)
+        message("-- ELFIO not found, fetching ELFIO...")
+        execute_process(COMMAND git submodule update --init)
+else ()
+        message("-- ELFIO found.")
+        add_subdirectory(third_party/ELFIO)
+endif()
 
-target_link_libraries(LibreCuda PRIVATE elfio)
+# create elf-parser library
+add_subdirectory(src)
+add_subdirectory(tests)
 
 # TODO: REMOVE WHEN CRUDE DEBUG MAIN GETS DELETED
-file(COPY ${CMAKE_SOURCE_DIR}/write_float.cubin DESTINATION ${CMAKE_BINARY_DIR})
\ No newline at end of file
+file(COPY ${CMAKE_SOURCE_DIR}/write_float.cubin DESTINATION ${CMAKE_BINARY_DIR}/bin)
```

Comment:
should be part of the test CMakeLists.txt file?
That of course intails moving all the trash .ptx and cubin files and the .sh file for building them to tests too.
Maybe consider adding tests/write_float/ subdirectory, which is the actual test module
whereas tests/ handles adding of test subdirectories and maybe functions for easy test creation in the future